File: app/src/main/java/com/blog/demo/system/ConnectivityManagerActivity.java
package com.blog.demo.system;

import android.app.Activity;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.TextView;

import com.blog.demo.R;

public class ConnectivityManagerActivity extends Activity {

    private ConnectivityManager connectivityManager;
    private TextView textView;

    private BroadcastReceiver mReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            if (intent.getAction().equals(ConnectivityManager.CONNECTIVITY_ACTION)) {
                NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
                if (networkInfo != null) {
                    textView.setText("Available = " + networkInfo.isAvailable() + "\n"
                            + "Connected = " + networkInfo.isConnected() + "\n"
                            + "Type = " + getType(networkInfo.getType()) + "\n"
                            + "Roaming = " + networkInfo.isRoaming() + "\n"
                            + "State = " + networkInfo.getState() + "\n"
                            + "DetailedState = " + networkInfo.getDetailedState() + "\n"
                            + "Failover = " + networkInfo.isFailover());
                } else {
                    textView.setText("no active network info");
                }
            }
        }
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_text);

        textView = findViewById(R.id.text_view);
        connectivityManager = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
    }

    @Override
    protected void onResume() {
        super.onResume();

        registerReceiver(mReceiver, new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION));
    }

    @Override
    protected void onPause() {
        super.onPause();

        unregisterReceiver(mReceiver);
    }

    private String getType(int type) {
        if (type == ConnectivityManager.TYPE_MOBILE) {
            return "MOBILE";
        } else if (type == ConnectivityManager.TYPE_WIFI) {
            return "WIFI";
        } else {
            return "unknow";
        }
    }

}

File: app/src/main/java/com/blog/demo/feature/ConnectivityManagerActivity.kt
package com.blog.demo.feature

import android.app.Activity
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.net.ConnectivityManager
import android.os.Bundle
import android.widget.TextView
import com.blog.demo.R

class ConnectivityManagerActivity : Activity() {

    private lateinit var textView: TextView
    private lateinit var connectivityManager: ConnectivityManager

    private val mReceiver: BroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            if (intent.action == ConnectivityManager.CONNECTIVITY_ACTION) {
                val networkInfo = connectivityManager!!.activeNetworkInfo
                if (networkInfo != null) {
                    textView.text = """
                    Available = ${networkInfo.isAvailable}
                    Connected = ${networkInfo.isConnected}
                    Type = ${getType(networkInfo.type)}
                    Roaming = ${networkInfo.isRoaming}
                    State = ${networkInfo.state}
                    DetailedState = ${networkInfo.detailedState}
                    Failover = ${networkInfo.isFailover}
                    """.trimIndent()
                } else {
                    textView.text = "no active network info"
                }
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_text)

        textView = findViewById(R.id.text_view)
        connectivityManager = getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager
    }

    override fun onResume() {
        super.onResume()
        registerReceiver(mReceiver, IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION))
    }

    override fun onPause() {
        super.onPause()
        unregisterReceiver(mReceiver)
    }

    private fun getType(type: Int): String {
        return if (type == ConnectivityManager.TYPE_MOBILE) {
            "MOBILE"
        } else if (type == ConnectivityManager.TYPE_WIFI) {
            "WIFI"
        } else {
            "unknow"
        }
    }

}